### PR TITLE
Sock update

### PR DESCRIPTION
**Pull request by AbelAguilera312**
Created the Update Sock Request.

### CODE REVIEW

File: api/types/stores.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ type SockStore interface {
 	CountSocks() (int, error)
 	GetSockByID(sockID int) (*Sock, error)
 	GetSockVariants(sockID int) ([]SockVariant, error)
+	UpdateSock(sockID int, sock Sock, variants []SockVariant) error
```
Comment (sebastian-nunez) [inline api/types/stores.go:20]: Add the signature for checking if a sock variant exists: `SockVariantExists(sockID int, size string) (bool, error)`

File: api/types/entities.go [hunk 2]
```diff
@@ -76,4 +76,10 @@ type OrderUpdateCreator struct {
 	FirstName string `json:"firstname"`
 	LastName  string `json:"lastname"`
 	Username  string `json:"username"`
-}
\ No newline at end of file
+}
+
+type UpdateSock struct {
```
Comment (sebastian-nunez) [inline api/types/entities.go:81]: 1. Remove `UpdateSock` from `entities.go` (not used)
2. Modify the `UpdateSockRequest` in `payloads.go` to look like this:

```go
type UpdateSockRequest struct {
	Sock     SockDTO          `json:"sock" validate:"required"`
	Variants []SockVariantDTO `json:"variants" validate:"required,dive"`
}
```

This is mainly so this endpoint is consistent with the create sock endpoint + you gain the ability to use some existing helper functions to convert from `SockDTO` to `Sock`.

File: api/services/inventory/sock_store.go [hunk 3]
```diff
@@ -204,3 +204,62 @@ func (s *SockStore) GetSockByID(sockID int) (*types.Sock, error) {
...
+				return fmt.Errorf("failed to insert variant: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// sockVariantExists checks if a sock variant exists for the given sock ID and size
+func (s *SockStore) sockVariantExists(sockID int, size string) (bool, error) {
```
Comment (sebastian-nunez) [inline api/services/inventory/sock_store.go:257]: This is useful functionality, so worthwhile making into an public method for the store. 

Add it to the `SockStore` interface: `SockVariantExists(sockID int, size string) (bool, error)`

File: api/services/inventory/sock_store.go [hunk 4]
```diff
@@ -204,3 +204,62 @@ func (s *SockStore) GetSockByID(sockID int) (*types.Sock, error) {
 
 	return &sock, nil
 }
+
+// UpdateSock updates an existing sock and its variants in the database
+func (s *SockStore) UpdateSock(sockID int, sock types.Sock, variants []types.SockVariant) error {
+	// Update the sock details
```
Comment (sebastian-nunez) [inline api/services/inventory/sock_store.go:210]: nit; you can remove all of the comments inside the function body, I think the code explains itself

File: api/services/inventory/sock_handler.go [hunk 5]
```diff
@@ -175,6 +177,72 @@ func (h *SockHandler) handleGetSockDetails(w http.ResponseWriter, r *http.Reques
 	utils.WriteJson(w, http.StatusOK, sock)
 }
 
+// @Summary Updates a sock and its variants
+// @Description Updates an existing sock's details and its variants.
+// @Tags Socks
```
Comment (sebastian-nunez) [inline api/services/inventory/sock_handler.go:182]: Should be part of the `Inventory` tag. Make sure to rerun `make swag`

File: api/services/inventory/sock_handler.go [hunk 6]
```diff
@@ -175,6 +177,72 @@ func (h *SockHandler) handleGetSockDetails(w http.ResponseWriter, r *http.Reques
...
+		utils.WriteError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	if err := utils.Validate.Struct(req); err != nil {
+		utils.WriteError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	// Convert the UpdateSockRequest to the types.Sock entity
```
Comment (sebastian-nunez) [inline api/services/inventory/sock_handler.go:222]: nit; the comments below are not really needed and don't add much context, so I would say remove them

File: api/services/inventory/sock_handler.go [hunk 7]
```diff
@@ -175,6 +177,72 @@ func (h *SockHandler) handleGetSockDetails(w http.ResponseWriter, r *http.Reques
...
+	sock := types.Sock{
```
Comment (sebastian-nunez) [inline api/services/inventory/sock_handler.go:223]: 1. Make the changes I suggested in the `entities.go` file (and updates for `UpdateSockRequest`)
2. With the new definition of `UpdateSockRequest`, the code should more or less look like this:

```go
    sock := toSock(req.Sock)
	variants := toSockVariantArray(req.Variants)

	if err := h.store.UpdateSock(sockID, sock, variants); err != nil {
		utils.WriteError(w, http.StatusInternalServerError, err)
		return
	}
//...
```

This way you reuse the help functions :)

**Review by sebastian-nunez — COMMENTED** [review]
Great work, some minor tweaks but looks good overall :)

File: api/types/stores.go (hunk 1 above)
Comment (AbelAguilera312) [reply api/types/stores.go:20]: Fixed!!

File: api/types/entities.go (hunk 2 above)
Comment (AbelAguilera312) [reply api/types/entities.go:81]: Ok understood

File: api/services/inventory/sock_store.go (hunk 3 above)
Comment (AbelAguilera312) [reply api/services/inventory/sock_store.go:257]: Got it! Thank you

File: api/services/inventory/sock_store.go (hunk 4 above)
Comment (AbelAguilera312) [reply api/services/inventory/sock_store.go:210]: Fixed!

File: api/services/inventory/sock_handler.go (hunk 6 above)
Comment (AbelAguilera312) [reply api/services/inventory/sock_handler.go:222]: Fixed!

File: api/services/inventory/sock_handler.go (hunk 7 above)
Comment (AbelAguilera312) [reply api/services/inventory/sock_handler.go:223]: Thank you! Fixed

**Review by sebastian-nunez — APPROVED** [review]
LGTM

**Pull request merged**